File: legacy-node/test/connectionNotifier.test.js
import test from "node:test";
import assert from "node:assert/strict";
import { buildConnectionReadyEmail } from "../src/connectionNotifier.js";

test("buildConnectionReadyEmail is notification-only without payment artifacts", () => {
  const { subject, text } = buildConnectionReadyEmail("SB-7K2M-9F3");
  assert.match(subject, /SB-7K2M-9F3/);
  assert.match(text, /Open SharingBridge/);
  assert.match(text, /do not send payment links/i);
  assert.doesNotMatch(text, /mailto:/i);
});

test("notifyConnectionReady sends webhook secret header when configured", async () => {
  const prevUrl = process.env.CONNECTION_NOTIFY_WEBHOOK_URL;
  const prevSecret = process.env.CONNECTION_NOTIFY_WEBHOOK_SECRET;
  process.env.CONNECTION_NOTIFY_WEBHOOK_URL = "http://notify.test/hook";
  process.env.CONNECTION_NOTIFY_WEBHOOK_SECRET = "test-secret";

  let capturedHeaders = null;
  const { notifyConnectionReady } = await import("../src/connectionNotifier.js");
  await notifyConnectionReady({
    orderCode: "SB-7K2M-9F3",
    recipientUserIds: ["alice"],
    lookupEmails: async () => ({ alice: "alice@example.com" }),
    fetchImpl: async (_url, init) => {
      capturedHeaders = init.headers;
      return { ok: true };
    }
  });

  assert.equal(capturedHeaders["x-webhook-secret"], "test-secret");

  if (prevUrl == null) {
    delete process.env.CONNECTION_NOTIFY_WEBHOOK_URL;
  } else {
    process.env.CONNECTION_NOTIFY_WEBHOOK_URL = prevUrl;
  }
  if (prevSecret == null) {
    delete process.env.CONNECTION_NOTIFY_WEBHOOK_SECRET;
  } else {
    process.env.CONNECTION_NOTIFY_WEBHOOK_SECRET = prevSecret;
  }
});

File: src/test/java/org/sharingbridge/integration/service/ConnectionNotifierTest.java
package org.sharingbridge.integration.service;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import com.fasterxml.jackson.databind.ObjectMapper;
import java.util.List;
import java.util.Map;
import java.util.concurrent.atomic.AtomicReference;
import org.junit.jupiter.api.Test;

class ConnectionNotifierTest {

    @Test
    void buildConnectionReadyEmailIsNotificationOnly() {
        Map<String, String> email = ConnectionNotifier.buildConnectionReadyEmail("SB-7K2M-9F3");
        assertTrue(email.get("subject").contains("SB-7K2M-9F3"));
        assertTrue(email.get("text").contains("Open SharingBridge"));
        assertTrue(email.get("text").toLowerCase().contains("do not send payment links"));
        assertFalse(email.get("text").toLowerCase().contains("mailto:"));
    }

    @Test
    void notifyConnectionReadySendsWebhookSecretHeaderWhenConfigured() {
        AtomicReference<Map<String, String>> captured = new AtomicReference<>();
        ConnectionNotifier notifier =
                new ConnectionNotifier(
                        (url, headers, body) -> {
                            captured.set(headers);
                            return 200;
                        },
                        new ObjectMapper(),
                        "http://notify.test/hook",
                        "test-secret");
        Map<String, Object> result =
                notifier.notifyConnectionReady(
                        "SB-7K2M-9F3",
                        List.of("alice"),
                        ids -> Map.of("alice", "alice@example.com"));
        assertEquals(true, result.get("sent"));
        assertEquals("test-secret", captured.get().get("x-webhook-secret"));
    }
}
